[노유빈] 11주차 과제 제출

YOOBINNOH opened the pull request:
.

YOOBINNOH committed:
[노유빈] 11주차 (P)60058번 과제 제출 합니다.

YOOBINNOH committed:
"[노유빈] 11주차 (P)60058번 과제 제출합니다."

YOOBINNOH committed:
[노유빈] 11주차 (P)60061번 제출합니다.

YOOBINNOH committed:
.DS_Store is banished!

donghoony commented on 11주차/(P)60058/(P)60058_python_yoobin.py:
```diff
@@ -0,0 +1,75 @@
+                return False
+        else:
+            a-=1
+            if a<0:
+                return False
```
11~12번 줄도 `a<0`의 조건에 `False`를 리턴하는데, if문 바깥에 쓰면 구문 하나로 정리할 수 있겠네요!

donghoony commented on 11주차/(P)60058/(P)60058_python_yoobin.py:
```diff
@@ -0,0 +1,75 @@
+    
+    u.append(p[0])
+    
+    for i in range(1,len(p)):
```
iterable object (반복적으로 접근이 가능한 객체)에 대해서, 인덱스가 필요하지 않다면 `for i in object`와 같이 반복을 할 수 있습니다!
위 코드는 `for i in p`로 수정하고, 32번 줄을 `u.append(i)`로 작성할 수 있겠네요!

donghoony commented on 11주차/(P)60058/(P)60058_python_yoobin.py:
```diff
@@ -0,0 +1,75 @@
+    
+    # 3. u가 올바른 괄호 문자열인 경우.
+    
+    if correct(u)!=False:
```
참과 거짓을 나타내는 `True`나 `False`를 파이썬에서 조건문으로 나타내고자 할 때에는, 
`if True`와 같이 사용합니다. 올바른 괄호 문자열인 경우이므로 `if correct(u):`라고만 작성해도 됩니다.

YOOBINNOH replied:
오호 감사합니다!

donghoony commented on 11주차/(P)60058/(P)60058_python_yoobin.py:
```diff
@@ -0,0 +1,75 @@
+    v = solution(v)
+    answer += v
+    answer += ")"
+    
```
4-4를 수행하지 않았습니다. 확인해 주세요!

YOOBINNOH replied:
그 u[0]이 한 문자가 아니라 문자열로 나와서 못했습니다..

donghoony replied:
u는 항상 문자열인데, 어떤 경우에 u[0]이 문자열로 나오나요?

YOOBINNOH replied:
u를 list 로 변환한뒤에 문제 조건에 따라 구현을 하려고 했는데 list로 바꾸니 u[0] 의 값이 한 문자가 아닌 여러 문자가 나와서 해결을 못했습니다.

donghoony replied:
위 코드에서는 u가 리스트가 아니에요! split에서 건네주는 건 그 함수 내의 u라서, 바깥 함수는 적용되지 않습니다

donghoony commented on 11주차/(P)60058/(P)60058_python_yoobin.py:
```diff
@@ -0,0 +1,75 @@
+    answer = ""
+    
+    
+    # 1. 입력이 빈 문자열인 경우
```
문제에서 주어진 Case별로 잘 구현하셨습니다 👍🏼

donghoony commented on 11주차/(P)60058/(P)60058_python_yoobin.py:
```diff
@@ -0,0 +1,75 @@
+        else:
+            a-=1
+            if a<0:
+                return False
```
나아가서, `(((`는 올바른 문자열인가요? 위 함수에서는 `(((`가 `None`을 리턴합니다.
`False`를 리턴하는 부분이 있다면, `True`를 리턴하는 부분도 구현해 주세요!

YOOBINNOH replied:
문제조건에 문자열 p를 이루는 '(' 와 ')' 의 개수는 항상 같습니다. 이게 있어서 생략했습니다!

donghoony replied:
그렇네요! 그렇다면 마지막에 항상 `return True`를 해 주면 되겠습니다. `False`만 반환하고, `True`를 반환하지 않네요

donghoony reviewed (changes requested):
정답에 거의 근접하셨습니다! 몇가지 피드백 함께 드릴게요~ 고생하셨습니다 😄

YOOBINNOH commented:
날짜: 2022.05.22 오전 02:41:45 (GMT+09:00)
제목: Re: [kth990303/BojCodingTestStudy] [노유빈] 11주차 과제 제출 (PR #59)

@donghoony requested changes on this pull request.
정답에 거의 근접하셨습니다! 몇가지 피드백 함께 드릴게요~ 고생하셨습니다 😄
In 11주차/(P)60058/(P)60058_python_yoobin.py:
> + if a<0: + return False
11~12번 줄도 a<0의 조건에 False를 리턴하는데, if문 바깥에 쓰면 구문 하나로 정리할 수 있겠네요!
In 11주차/(P)60058/(P)60058_python_yoobin.py:
> + else: + a-=1 + if a<0: + return False + + +# u,v 로 나누어 주는 함수 + +def split(p): + u = [ ] + + count = 0 + + u.append(p[0]) + + for i in range(1,len(p)):
iterable object (반복적으로 접근이 가능한 객체)에 대해서, 인덱스가 필요하지 않다면 for i in object와 같이 반복을 할 수 있습니다!
위 코드는 for i in p로 수정하고, 32번 줄을 u.append(i)로 작성할 수 있겠네요!
In 11주차/(P)60058/(P)60058_python_yoobin.py:
> + if p == "": + return "" + + + + # 2. u, v 로 분리 후 값 대입 + + + u = p[0:split(p)+1] + v = p[split(p)+1:] + + + + # 3. u가 올바른 괄호 문자열인 경우. + + if correct(u)!=False:
참과 거짓을 나타내는 True나 False를 파이썬에서 조건문으로 나타내고자 할 때에는,
if True와 같이 사용합니다. 올바른 괄호 문자열인 경우이므로 if correct(u):라고만 작성해도 됩니다.
In 11주차/(P)60058/(P)60058_python_yoobin.py:
> + + + # 3. u가 올바른 괄호 문자열인 경우. + + if correct(u)!=False: + return u+solution(v) + + + + + # 4. u가 올바른 괄호 문자열이 아닌 상황. + answer = "(" + v = solution(v) + answer += v + answer += ")" +
4-4를 수행하지 않았습니다. 확인해 주세요!
In 11주차/(P)60058/(P)60058_python_yoobin.py:
> + for i in range(1,len(p)): + if u.count('(') == u.count(')'): + break + else: + u.append(p[i]) + count += 1 + + return count + + + +def solution(p): + answer = "" + + + # 1. 입력이 빈 문자열인 경우
문제에서 주어진 Case별로 잘 구현하셨습니다 👍🏼
In 11주차/(P)60058/(P)60058_python_yoobin.py:
> +# 괄호 변환 : 실패, 거의 다 했다고 생각을 했는데 u[0] 값이 한 개가 아닌 복수 값이 나오며 제대로 되지 않았다. split 함수에서 return 값에 문제가 있던 것 같다. + + +# 올바른 문자열인지 check 하는 함수 + +def correct(u): + a=0 + for i in range(0,len(u)): + if u[i]=="(": + a+=1 + if a<0: + return False + else: + a-=1 + if a<0: + return False
나아가서, (((는 올바른 문자열인가요? 위 함수에서는 (((도 None을 리턴합니다.
False를 리턴하는 부분이 있다면, True를 리턴하는 부분도 구현해 주세요!
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

kth990303 commented:
위 코멘트는 뭐죠?ㅋㅋㅋ
